addons/stock/models/stock_move.py

```diff
@@ -1450,21 +1450,19 @@ def _action_confirm(self, merge=True, merge_into=False):
 
         # create procurements for make to order moves
         procurement_requests = []
-        procurement_quantities = move_create_proc._prepare_procurement_qty()
-        for move, quantity in zip(move_create_proc, procurement_quantities):
+        quantities = move_create_proc._prepare_procurement_qty()
+        for move, quantity in zip(move_create_proc, quantities):
+            if float_is_zero(quantity, precision_rounding=move.product_uom.rounding):
+                move_to_confirm |= move
             values = move._prepare_procurement_values()
             origin = move._prepare_procurement_origin()
             procurement_requests.append(self.env['procurement.group'].Procurement(
                 move.product_id, quantity, move.product_uom,
                 move.location_id, move.rule_id and move.rule_id.name or "/",
                 origin, move.company_id, values))
+            move_waiting |= move
```

Comment:
Maybe leave `move_waiting` as an `OrderedSet()` at this point for performance reasons at this point (i.e. avoid the `|=`) and then do the browse right before you need it